From c1f107c24cec56f6305aad442bf6f51d1ff58fb9 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:53:23 -0400
Subject: [net] sunrpc: Fix a module reference issue in rpcsec_gss

Message-id: <1403798077-28799-5-git-send-email-steved@redhat.com>
Patchwork-id: 84879
O-Subject: [RHEL7.1 PATCH 04/78] SUNRPC: Fix a module reference issue in rpcsec_gss
Bugzilla: 1113248
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Trond Myklebust <trond.myklebust@primarydata.com>

We're not taking a reference in the case where _gss_mech_get_by_pseudoflavor
loops without finding the correct rpcsec_gss flavour, so why are we
releasing it?

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Upsteam Id: 7a9a7b774fd543467313894fe53bc7dcc47d5708
Bugzilla: 1113248
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sunrpc/auth_gss/gss_mech_switch.c b/net/sunrpc/auth_gss/gss_mech_switch.c
index 27ce262..92d5ab9 100644
--- a/net/sunrpc/auth_gss/gss_mech_switch.c
+++ b/net/sunrpc/auth_gss/gss_mech_switch.c
@@ -218,10 +218,8 @@ static struct gss_api_mech *_gss_mech_get_by_pseudoflavor(u32 pseudoflavor)
 
  spin_lock(&registered_mechs_lock);
  list_for_each_entry(pos, &registered_mechs, gm_list) {
-  if (!mech_supports_pseudoflavor(pos, pseudoflavor)) {
-   module_put(pos->gm_owner);
+  if (!mech_supports_pseudoflavor(pos, pseudoflavor))
    continue;
-  }
   if (try_module_get(pos->gm_owner))
    gm = pos;
   break;
-- 
1.7.1